server/src/com/cloud/vm/snapshot/VMSnapshotManagerImpl.java

```diff
@@ -707,16 +802,41 @@ private UserVm orchestrateRevertToVMSnapshot(Long vmSnapshotId) throws Insuffici
             throw new InvalidParameterValueException("There is other active vm snapshot tasks on the instance, please try again later");
         }
 
+        revertUserVmDetailsFromVmSnapshot(userVm, vmSnapshotVo);
+
         try {
             VMSnapshotStrategy strategy = findVMSnapshotStrategy(vmSnapshotVo);
             strategy.revertVMSnapshot(vmSnapshotVo);
+            updateUserVmServiceOffering(userVm, vmSnapshotVo);
```

Comment:
Service offering and details update should be done in a transaction once the snapshot revert completes successfully provided there is no agent call happening as part of _userVmManager.upgradeVirtualMachine()